File: UnitC/ch06/ex-6.243.jl
"""
chapter:ch06
page 499  exercise  enhance  anti-drepression
data:StressedMice

 
ttest results:
point estimate:          -207.0
 95% confidence interval: (-215.6, -198.4)
 outcome with 95% confidence: reject h_0
"""


## 1 load package
include("../../utils.jl")
using UnicodePlots

## 2. load data
desc = Lock5Table(499, "StressedMice", "exercise  enhance  anti-drepression", ["Time", "Environment"])
# 1:enriched, 2:standard
df = @pipe load_csv(desc.name, false) |> select(_, desc.feature) |> groupby(_, desc.feature[2])

data = [df[1].Time, df[2].Time]

##3.  desc table
Group = @pipe keys(df) .|> values(_)[1]
summary_df = DataFrame(Group=Group,
    n=size.(data, 1),
    Mean=mean.(data),
    Stddev=std.(data)
)
@pt summary_df
#= 
┌──────────┬───────┬─────────┬─────────┐
│    Group │     n │    Mean │  Stddev │
│ String15 │ Int64 │ Float64 │ Float64 │
├──────────┼───────┼─────────┼─────────┤
│ Enriched │     7 │ 231.714 │ 71.2267 │
│ Standard │     7 │ 438.714 │ 37.6816 │
└──────────┴───────┴─────────┴─────────┘
=#

## 4. box-plot
UnicodePlots.boxplot(Group, data, title="time in dark place", xlabel="time", ylabel="group")

#= 
  time in dark place             
                  ┌                                        ┐ 
                      ╷    ┌───┬──┐         ╷                
         Enriched     ├────┤   │  ├─────────┤                
   group              ╵    └───┴──┘         ╵                
                                               ╷ ┌──┬──┐ ╷   
         Standard                              ├─┤  │  ├─┤   
                                               ╵ └──┴──┘ ╵   
                  └                                        ┘ 
                   100               300                500  
                                     time           
=#


## 5. pair ttest
nx, mx, vx = summary_df[1, 2:4]
ny, my, vy = summary_df[2, 2:4]
EqualVarianceTTest(Int(nx), Int(ny), mx, my, vx, vy)

#= 
Two sample t-test (equal variance)
----------------------------------
Population details:
    parameter of interest:   Mean difference
    value under h_0:         0
    point estimate:          -207.0
    95% confidence interval: (-215.6, -198.4)

Test summary:
    outcome with 95% confidence: reject h_0
    two-sided p-value:           <1e-14

Details:
    number of observations:   [7,7]
    t-statistic:              -52.47940404269733
    degrees of freedom:       12
    empirical standard error: 3.9444045483364185
=#

File: marimo/ch06/ex-6.243.py
import marimo

__generated_with = "0.24.0"
app = marimo.App(width="medium")


@app.cell
def _():
    import marimo as mo
    import sys
    from pathlib import Path
    import numpy as np
    import polars as pl
    import matplotlib.pyplot as plt
    from scipy import stats
    from great_tables import GT
    _p = Path(mo.notebook_location()).resolve().parent
    while _p != _p.parent and not (_p / "lock5.py").exists():
        _p = _p.parent
    if str(_p) not in sys.path:
        sys.path.insert(0, str(_p))
    import lock5
    return mo, sys, Path, np, pl, plt, stats, GT, lock5


@app.cell
def _(mo):
    mo.md(r"""# Ex 6.243 — Exercise enhances anti-depression? (StressedMice)

Port of `UnitC/ch06/ex-6.243.jl`: compare time in the dark between `Enriched` and
`Standard` environments, then run a two-sample t-test.""")
    return


@app.cell
def _(lock5):
    desc = lock5.Lock5Table(499, "StressedMice", "exercise enhance anti-drepression", ["Time", "Environment"])
    df = lock5.load_csv(desc.name, drop=False).select(desc.feature)
    group, data, summary = lock5.group_summary(df, desc.feature[1], desc.feature[0])
    return desc, df, group, data, summary


@app.cell
def _(GT, mo, summary):
    mo.Html(GT(summary).tab_header(title="Time in dark by environment").fmt_number(columns=["Mean", "Stddev"], decimals=4).as_raw_html())
    return


@app.cell
def _(data, group, lock5):
    fig, ax = lock5.grouped_boxplot(group, data, title="time in dark place", xlabel="time", ylabel="group")
    fig
    return fig, ax


@app.cell
def _(lock5, mo, summary):
    res = lock5.pair_ttest(summary)
    mo.Html(lock5.gt_ttest(res).as_raw_html())
    return res


if __name__ == "__main__":
    app.run()
